torchx/lib/torchx/backend.ex

```diff
@@ -748,6 +748,48 @@ defmodule Torchx.Backend do
     |> to_nx(out)
   end
 
+  # shape: shape, type: type
+  @impl true
+  def select(out, %{shape: {}} = pred, on_true, on_false) do
+    if to_number(pred) == 0 do
+      on_false
+      |> from_nx()
+      |> Torchx.broadcast_to(out.shape)
+      |> to_nx(out)
+    else
+      on_true
+      |> from_nx()
+      |> Torchx.broadcast_to(out.shape)
+      |> to_nx(out)
+    end
+  end
+
+  def select(%T{type: out_type} = out, pred, on_true, on_false) do
+    on_true_torch =
+      on_true
+      |> from_nx()
+
+    on_false_torch =
+      on_false
+      |> from_nx()
+
+    if out_type == :bool do
+      pred_torch =
+        pred
+        |> from_nx()
```

Comment:
Minor nitpick: we usually avoid `|>` if there is only one `|>`:

```suggestion
    on_true_torch = from_nx(on_true)
    on_false_torch = from_nx(on_false)

    if out_type == :bool do
      pred_torch = from_nx(pred)
```

:)